frontend/src/views/external-api.js

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+
+const ExternalApi = () => {
+  const [response, setResponse] = useState("");
+  const [token, setToken] = useState("");
+  const serverUrl = process.env.REACT_APP_SERVER_URL;
+
+  const { getAccessTokenSilently } = useAuth0();
+
+  const refreshToken = async () => {
+    try {
+      const accessToken = await getAccessTokenSilently();
+      setToken(accessToken);
+    } catch (error) {
+      console.log(error.message);
+    }
+  };
+
+  useEffect(() => {
+    refreshToken();
+  });
```

Comment:
Without the dependencies argument, I think this useEffect will be triggered by each update of the component using the hook.
You should double-check. If it's being hit multiple times by a single component, you should add an [] as the second argument:
```javascript
useEffect(() => {
  refreshToken();
}, []);
```

But I'm not 100% sure if there's a difference between an empty dependencies array vs. none at all, but my understanding of the documentation for the hook suggests that there is.

Reply:
`refreshToken` is just a wrapper around auth0's `getAccessTokenSilently`, which does not send a HTTP request every time it's called. It caches the token somewhere. So I don't see the need of reducing refreshToken being called. What do you think?